widget/discrepancy-note/dn-widget.js

```diff
@@ -531,35 +530,33 @@ class Comment extends Widget {
     }
 
     /**
-     * Sets users, currentUser, and usersOptionsHtml global variables (once for all dn widgets);
+     * Sets users and currentUser global variables (once for all dn widgets);
      * 
      * @param {boolean=} readOnly 
      */
     _setUserOptions( readOnly ) {
-        if ( !usersOptionsHtml ) {
-            const disabled = readOnly ? 'disabled' : '';
-            const defaultAssignee = this.defaultAssignee;
-            try {
-                const userNodes = this.options.helpers.evaluate( 'instance("_users")/root/item', 'nodes', null, null, true );
-
-                // doing this in 2 steps as it is likely useful later on to store the users array separately.
-                users = userNodes.map( item => ( {
-                    firstName: item.querySelector( 'first_name' ).textContent,
-                    lastName: item.querySelector( 'last_name' ).textContent,
-                    userName: item.querySelector( 'user_name' ).textContent
-                } ) );
-                usersOptionsHtml = `<option value="" ${disabled}></option>${users.map( user => {
-                    const readableName = `${user.firstName} ${user.lastName} (${user.userName})`;
-                    const selected = user.userName === defaultAssignee ? ' selected ' : '';
-                    return `<option value="${user.userName}"${selected}${disabled}>${readableName}</option>`;
-                } )}`;
-
-                const currentUsernameNode = this.options.helpers.evaluate( 'instance("_users")/root/item[@current]/user_name', 'node', null, null, true );
-                currentUser = currentUsernameNode ? currentUsernameNode.textContent : null;
-            } catch ( e ) {
-                //users = [];
-                console.error( e );
-            }
+        const disabled = readOnly ? 'disabled' : '';
+        const defaultAssignee = this.defaultAssignee;
+        try {
+            const userNodes = this.options.helpers.evaluate( 'instance("_users")/root/item', 'nodes', null, null, true );
+
+            // doing this in 2 steps as it is likely useful later on to store the users array separately.
+            users = userNodes.map( item => ( {
```

Comment:
I wonder if it would be good to make the `users` variable global, since it may take some time for long forms and is the same for each widget. So for a form with a 100 questions and a 100 discrepancy notes widgets, it would do this 100 times *upon initial form load*

If we're doing that, how about taking the code that creates `this.usersOptionHtml out and just do this in a separate function called in init, or even just do it on the fly (inline?) in `renderForm()`. The performance won't matter as much because only one widget can be rendered at a time anyway.

Reply:
got it, code updated